### PR TITLE
Set page title for each sidebar section through App management pages

### LINKED ISSUE #33852: [BITV] App management - Window title must change when switching sidebar sections

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
Adapt router.js for setting document title

Fixes https://github.com/nextcloud/server/issues/33852

### CODE REVIEW

File: apps/settings/src/router.js [hunk 1]
```diff
@@ -53,10 +53,26 @@ export default new Router({
...
+			},
 			children: [
 				{
 					path: ':selectedGroup',
 					name: 'group',
+					meta: {
+						title: (to) => {
+							if (to.params.selectedGroup === 'admin' ) {
+								return t('settings', 'Admins')
+							} else if (to.params.selectedGroup === 'disabled') {
```
Comment (ChristophWurst) [inline apps/settings/src/router.js:69]: no need for else-if or else if all if branches return

**Review by ChristophWurst — APPROVED** [review]

File: apps/settings/src/router.js [hunk 2]
```diff
@@ -53,10 +53,26 @@ export default new Router({
...
+					return t('settings', 'Active users')
+				},
+			},
 			children: [
 				{
 					path: ':selectedGroup',
 					name: 'group',
+					meta: {
+						title: (to) => {
+							if (to.params.selectedGroup === 'admin' ) {
```
Comment (ChristophWurst) [inline apps/settings/src/router.js:67]:
```suggestion
							if (to.params.selectedGroup === 'admin') {
```
to make eslint happy

File: apps/settings/src/router.js (hunk 1 above)
Comment (JuliaKirschenheuter) [reply apps/settings/src/router.js:69]: i need to differentiate between 'admin' and 'disabled' and i think i need "else if" in this case, no?

**Review by ChristophWurst — APPROVED** [review]
:+1: still

File: apps/settings/src/router.js [hunk 3]
```diff
@@ -53,10 +53,25 @@ export default new Router({
 			component: Users,
 			props: true,
 			name: 'users',
+			meta: {
+				title: () => {
+					return t('settings', 'Active users')
+				},
```
Comment (skjnldsv) [inline apps/settings/src/router.js:59]: Does it need to be a function?

```suggestion
				title: t('settings', 'Active users')
```

File: apps/settings/src/router.js [hunk 4]
```diff
@@ -83,3 +98,14 @@ export default new Router({
 		},
 	],
 })
+
+router.afterEach(async (to) => {
+	const metaTitle = await to.meta.title?.(to)
```
Comment (skjnldsv) [inline apps/settings/src/router.js:103]: There is no async methods in above titles ?

Comment (JuliaKirschenheuter) [reply apps/settings/src/router.js]: yes, already seen an fixed. Thnaks!

File: apps/settings/src/router.js [hunk 5]
```diff
@@ -53,10 +53,25 @@ export default new Router({
...
 			children: [
 				{
 					path: ':selectedGroup',
 					name: 'group',
+					meta: {
+						title: (to) => {
+							if (to.params.selectedGroup === 'admin') {
+								return t('settings', 'Admins')
+							} if (to.params.selectedGroup === 'disabled') {
+								return t('settings', 'Disabled users')
```
Comment (Pytal) [inline apps/settings/src/router.js:70]:
```suggestion
							}
							if (to.params.selectedGroup === 'disabled') {
								return t('settings', 'Disabled users')
```

**Review by Pytal — APPROVED** [review]

**Comment by szaimen** [comment]
CI failure unrelated

**Pull request merged**